### PR TITLE
feat(server): implement json.debug command (#104)

### PR DESCRIPTION
**Pull request by iko1**
Signed-off-by: iko1 <me@remotecpp.dev>

### CODE REVIEW

File: src/server/json_family.cc [hunk 1]
```diff
@@ -286,6 +287,36 @@ string ConvertToJsonPointer(string_view json_path) {
...
+  } else {
+    res += 1;
+  }
+}
+
+size_t CountJsonFields(const json& j) {
+  size_t res = 0;
+  CountJsonFieldsImpl(j, res);
+  return res;
+}
```
Comment (dranikpg) [inline src/server/json_family.cc:318]: Just a side note, if you always return the additional size, you don't need a separate impl function

File: src/server/json_family.cc [hunk 2]
```diff
@@ -872,8 +903,77 @@ OpResult<vector<OptString>> OpMGet(const OpArgs& op_args, const vector<string_vi
...
+    return result.status();
+  }
+
+  vector<OptSizeT> vec;
+  auto cb = [&vec](const string_view& path, const json& val) {
+    if (val.is_array() || val.is_object()) {
+      vec.emplace_back(CountJsonFields(val));
+    } else {
+      vec.emplace_back(1);
+    }
```
Comment (dranikpg) [inline src/server/json_family.cc:920]: Doesn't CountJsonFields already return the correct value for non object and non array types?

Comment (iko1) [reply src/server/json_family.cc]: According to your suggestion, I amended the loop body.

File: src/server/json_family.cc (hunk 1 above)
Comment (iko1) [reply src/server/json_family.cc]: According to your suggestion, I removed the separate implementation function, Thanks!

**Review by dranikpg — APPROVED** [review]

File: src/server/json_family.cc [hunk 3]
```diff
@@ -905,8 +933,73 @@ OpResult<vector<OptString>> OpMGet(const OpArgs& op_args, const vector<string_vi
...
+  };
+
+  expression.evaluate(*result, cb);
+  return vec;
+}
+
 }  // namespace
 
+void JsonFamily::Debug(CmdArgList args, ConnectionContext* cntx) {
+  function<decltype(OpFields)> func;
```
Comment (romange) [inline src/server/json_family.cc:956]: I am not sure I understand why you need this "func" complication.
why not just call `OpFields` at line 990?

**Review by romange — COMMENTED** [review]
LGTM with a question

**Review by romange — APPROVED** [review]

**Pull request merged**